### PR TITLE
Added tests for libgit2 and an error throw

### PR DESCRIPTION
**Pull request by kshyatt**
I think we ought to have a `length` check in that `getindex` since it's not an `unsafe_getindex` (although when I tried the unsafe thing, nothing bad happened, it could conceivably in the future).

Otherwise, I added in a few tests for untested methods/error throws.

### CODE REVIEW

File: test/libgit2.jl [hunk 1]
```diff
@@ -129,9 +130,18 @@ mktempdir() do dir
...
                 @test isdir(path)
                 @test isfile(joinpath(path, LibGit2.Consts.HEAD_FILE))
+                @test LibGit2.isattached(repo)
             finally
                 finalize(repo)
             end
+
+            path = joinpath("garbagefakery", "Example.Bare")
+            try
+                LibGit2.GitRepo(path)
```
Comment (tkelman) [inline test/libgit2.jl]: add something like `error("unexpected")` here otherwise if it succeeds the tests will not trigger

**Comment by kshyatt** [comment]
@tkelman Better?

**Comment by kshyatt** [comment]
Also @wildart

**Comment by wildart** [comment]
:+1:

**Comment by kshyatt** [comment]
Do I need to squash or is this good to merge?

**Pull request merged**